source/index-management/tests/README.rst

```diff
@@ -45,4 +44,154 @@ Tests
+   2. An index with the ``name`` of ``test-search-index-2`` is present and index has a field ``queryable`` with the value of ``true``.
+#. For each ``index`` in ``indexDefinitions``
+   1. Find the matching index definition in ``indexes`` by matching on ``index.name``.  If no index exists, raise an error.
+   2. Assert that the matching index ``mappings``, whose value is ``{ dynamic: false }``
+
+Case 3: Driver can successfully drop search indexes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+#. Create a collection with a randomly generated name (referred to as ``coll0``).
+#. Create a new search index on ``coll0`` with the following definition:
+
+  .. code:: typescript
+
+    {
+      name: 'test-search-index',
+      definition: {
+        mappings: { dynamic: false }
+      }
+    }
+
+#. Assert that the command returns the name of the index: ``"test-search-index"``.
+#. Run ``coll0.listSearchIndexes()`` repeatedly every 5 seconds until the following condition is satisfied:
+   1. An index with the ``name`` of ``test-search-index`` is present and index has a field ``queryable`` with the value of ``true``.
+
+#. Run a ``dropSearchIndexes`` on ``coll0``, using ``test-search-index`` for the name.
+#. Run ``coll0.listSearchIndexes()`` repeatedly every 5 seconds until ``listSearchIndexes`` returns an empty array.
+
+This test fails if it times out waiting for the deletion to succeed.
+
+Case 4: Driver can update a search index
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+#. Create a collection with a randomly generated name (referred to as ``coll0``).
+#. Create a new search index on ``coll0`` with the following definition:
+
+  .. code:: typescript
+
+    {
+      name: 'test-search-index',
+      definition: {
+        mappings: { dynamic: false }
+      }
+    }
+
+#. Assert that the command returns the name of the index: ``"test-search-index"``.
+#. Run ``coll0.listSearchIndexes()`` repeatedly every 5 seconds until the following condition is satisfied:
+   1. An index with the ``name`` of ``test-search-index`` is present and index has a field ``queryable`` with the value of ``true``.
+
+#. Run a ``updateSearchIndex`` on ``coll0``, using the following definition.
+  
+  .. code:: typescript
+
+    {
+      name: 'test-search-index',
+      definition: {
+        mappings: { dynamic: true }
+      }
+    }
+
+#. Assert that the command does not error and the server responds with a success.
+#. Run ``coll0.listSearchIndexes()`` repeatedly every 5 seconds until the following condition is satisfied:
+   1. An index with the ``name`` of ``test-search-index`` is present.  This index is referred to as ``index``.
+   2. The index has a field ``queryable`` with a value of ``true`` and has a field ``status`` with the value of ``READY``.
+  
+#. Assert that an index is present with the name ``test-search-index`` and the definition has a
+  property ``mappings``, whose value is ``{ dynamic: true }``
```

Comment:
```suggestion
  property ``latestDefinition``, whose value is ``{ 'mappings' : { 'dynamic' : true } }``
```

Suggest updating assert to use `latestDefinition` to match returned document. Here is an example of the documents returned in the C driver testing:

```
{ "id" : "64c1651aa199d3199e27ae1a", "name" : "test-search-index", "status" : "PENDING", "queryable" : false, "latestDefinition" : { "mappings" : { "dynamic" : false } } }
```